docs/design/round-trip-latency/round-trip-latency.md

```diff
@@ -2,83 +2,82 @@
 
 ## Current implementation
 
-In current implementation we measure latency from Switch A to Switch B by discovery packets.
+In the current implementation, we measure latency from Switch A to Switch B using discovery packets.
 ![Current latency implementation](current_discovery.png "Current latency implementation")
 
 ## Round trip latency design with NoviFlow switches
 
 The new way to measure round trip latency:
 
-1) Send a discovery as a packet_out to switch A
-2) Switch A put a timestamp t1 to the discovery packet and send it to switch B
+1) Send a discovery packet as a packet_out to switch A.
+2) Switch A puts a timestamp t1 to the discovery packet and sends it to Switch B.
 3) Switch B receives the discovery packet and copies it. 
-After copying we got two packets - P1 (for discovery) and P2 (for round trip latency).
-Switch B has different actions for where two copies: 
-    * P1 will be send to the controller
-    * P2 will be modified (UDP port will be changed) and send back to switch A (via group table)
-4) Switch A receives his own message, add a timestamp t2 to it and send it to a controller.
-5) Controller use the discovery packet from Switch B for ISL discovery purposes and store ISL in DB (without latency).
-6) Controller use the discovery packet from Switch A for a roundtrip latency (t2-t1) measurement purposes:
-    * Save ISL latency in DB
-    * Send latency stats to OTSDB 
+After copying, we got two packets: P1 (for discovery) and P2 (for round trip latency).
+Switch B has different actions for these two copies: 
+    * P1 will be sent to the controller,
+    * P2 will be modified (UDP port will be changed) and send back to Switch A (via group table).
+4) Switch A receives his own message, adds a timestamp t2 to it, and sends it to the controller.
+5) The controller uses the discovery packet from Switch B for ISL discovery purposes and stores ISL in DB (without latency).
+6) The controller uses the discovery packet from Switch A for a round trip latency (t2-t1) measurement purposes:
+    * Saves ISL latency in DB,
+    * Sends latency stats to OTSDB. 
 
 ![Round trip latency design](new_discovery_noviflow_with_groups_support.png "Round trip latency design")
 
 ## Round trip latency design with NOT NoviFlow switch
```

Comment:
Round trip latency design with **NOT** NoviFlow switch

Non-NoviFlow

Reply:
Updated